File: js/caesar-cipher.js
function caesarCipher(s, kk) {
    console.log(kk); 
    console.log(s);    
    
    var k = kk % 26;
    console.log(k);

    if (k < 0) 
        k = k + 26;

    console.log(k);
    
    let result = ""

    for (let i=0; i<s.length ;i++)
        {
            let char = s[i];
            let code = s.charCodeAt(i);
            
            if(code >= 65 && code <= 90)
            {
                let shiftedCode = ((code - 65 + k) % 26) + 65;
                result += String.fromCharCode(shiftedCode);
            } else if (code >= 97 && code <= 122) { 
                let shiftedCode = ((code - 97 + k) % 26) + 97;
                result += String.fromCharCode(shiftedCode);
            } else { 
                result += char;
            }                          
        }
    return result;
}

//console.log(caesarCipher("middle-Outza", 2));
console.log(caesarCipher("middle-Outvwxyza", -56));

File: c#/hr/caesarcipher.cs
static string caesarCipher(string s, int kk)    {
        Console.WriteLine(kk);
        Console.WriteLine(s);

        var k = kk % 26;
        Console.WriteLine(k);

        if(k<0)
            k = k + 26;

        Console.WriteLine(k);

        char[] arr;
        arr = s.ToCharArray(0, s.Length);
        //Console.WriteLine("arr: "+new string(arr));
        for(int i=0; i<arr.Count();i++)
        {
            int t = arr[i];
            char c = (char)(t + k);

            if(t+k<65 || t+k>122 || (t+k>90 && t+k<97)) Console.Write($"! i: {i}, char: {(char)t}, t: {t}, c: {c} - ");

            if(t + k > 26 && ((t>64 && t<91) ||(t>96 && t<123)) )
            {
                Console.WriteLine("t+k > 26 ");
                //Console.WriteLine(((char)t).ToString()+" "+t.ToString()+" -> "+c);
                arr[i] = c;
            }
            
            if(t + k < 27 &&  ((t>64 && t<91) ||(t>96 && t<123)) )
            {
                Console.WriteLine("t+k < 27 ");
                ////Console.WriteLine(t.ToString()+" >> "+c);
                //Console.WriteLine(((char)t).ToString()+" "+t.ToString()+" -> "+c);
                arr[i] = c;                
            }            
                
        }        
        string result = new string(arr);
        Console.WriteLine("");
        return result;
    }
//middle-Outza
Console.WriteLine(caesarCipher("middle-Outvwxyza", -56));